gui/webdisplay/src/RWebDisplayHandle.cxx

```diff
@@ -241,6 +241,49 @@ RWebDisplayHandle::BrowserCreator::Display(const RWebDisplayArgs &args)
 
    std::string rmdir = MakeProfile(exec, args.IsBatchMode() || args.IsHeadless());
 
+   // these are secret parameters, hide them in temp file
+   if ((url.find("token=") || url.find("key=")) && !args.IsBatchMode() && !args.IsHeadless()) {
+      gRandom->SetSeed(0);
+
+      std::string tmpfile = gSystem->TempDirectory();
+
+#ifdef _MSC_VER
+      char slash = '\\';
+#else
+      char slash = '/';
+#endif
+      if (!tmpfile.empty() && (tmpfile[tmpfile.length()-1] != slash))
+         tmpfile += slash;
+      tmpfile += "root_start_"s + std::to_string(gRandom->Integer(0x100000)) + ".html";
```

Comment:
Shouldn't we use `TSystem::TempFileName`? On UNIX, it uses `mkstemp`, which makes sure that there is no race and it sets the file permission of the new file to `0600`, which is what we want.

Reply:
There is `mkstemps` for that. (We could extend `TSystem::TempFileName` to use it)